pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -84,13 +86,64 @@ type Client struct {
 
 	// The gateway-api client we will use to access objects
 	gatewayAPIClient gatewayapiclient.Interface
+
+	// initialSync is set to true after performing an initial processing of all objects.
+	initialSync *atomic.Bool
 }
 
 var _ model.ConfigStoreCache = &Client{}
 
+func New(client kube.Client, revision, domainSuffix string) (model.ConfigStoreCache, error) {
+	schemas := collections.Pilot
+	if features.EnableServiceApis {
+		schemas = collections.PilotServiceApi
+	}
+	return NewForSchemas(context.Background(), client, revision, domainSuffix, schemas)
+}
+
+func NewForSchemas(ctx context.Context, client kube.Client, revision, domainSuffix string, schemas collection.Schemas) (model.ConfigStoreCache, error) {
+	out := &Client{
+		domainSuffix:     domainSuffix,
+		schemas:          schemas,
+		revision:         revision,
+		queue:            queue.NewQueue(1 * time.Second),
+		kinds:            map[config.GroupVersionKind]*cacheHandler{},
+		istioClient:      client.Istio(),
+		gatewayAPIClient: client.GatewayAPI(),
```

Comment:
Is n't this needed only when we enable serviceApis?

Reply:
I donot change this part, just move the construction function up

Reply:
Whatever this is a good catch, will fix it.

Reply:
#33258